Fix test failure on Windows due to path separator mismatch

### Describe the bug

When running the test suite on a Windows machine, the tests fail because of path separator mismatches. 

Specifically, in `src/index.test.ts`, the mock for `existsSync` expects POSIX-style paths (e.g. `/mock-workspace/node_modules/.bin/oxlint`), but on Windows, `path.join` produces paths with backslashes (e.g. `\mock-workspace\node_modules\.bin\oxlint`). This causes the extension activation test to fail on Windows environments.

### Steps to reproduce

1. Clone the repository on a Windows machine.
2. Run `pnpm install`
3. Run `pnpm exec vp test`
4. Observe the failure:
```text
 FAIL  src/index.test.ts > extension activation > registers both clients with the expected initialization options
AssertionError: expected { run: { …(3) }, debug: { …(3) } } to match object { run: { …(2) } }

--- a/src/index.test.ts
+++ b/src/index.test.ts
@@ -1,4 +1,5 @@
 import { beforeEach, describe, expect, it, vi } from "vite-plus/test";
+import { join } from "path";
 
 const mocks = vi.hoisted(() => {
   const configurationValues: Record<string, Record<string, unknown>> = {};
@@ -182,10 +183,10 @@ describe("extension activation", () => {
       enable: true,
       binPath: "/mock/bin/oxfmt",
     };
-    mocks.existsSync.mockImplementation(
-      (path: string) =>
-        path === "/mock-workspace/node_modules/.bin/oxlint" || path === "/mock/bin/oxfmt",
-    );
+    mocks.existsSync.mockImplementation((path: string) => {
+      const p = path.replace(/\\/g, "/");
+      return p === "/mock-workspace/node_modules/.bin/oxlint" || p === "/mock/bin/oxfmt";
+    });
 
     const { activate } = await import("./index");
     const context = { subscriptions: [] as unknown[] };
@@ -204,7 +205,7 @@ describe("extension activation", () => {
     const [oxlintClient, oxfmtClient] = mocks.createdClients;
     expect(oxlintClient.serverOptions).toMatchObject({
       run: {
-        command: "/mock-workspace/node_modules/.bin/oxlint",
+        command: join("/mock-workspace", "node_modules", ".bin", "oxlint"),
         args: ["--lsp"],
       },
     });

